scipy/stats/_multicomp.py

```diff
@@ -0,0 +1,286 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from scipy import stats
+from scipy.stats._qmc import check_random_state
+from scipy.stats._stats_py import _var
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+    from scipy._lib._util import SeedType
+    from typing import Literal, Sequence  # noqa: UP035
+
+
+__all__ = [
+    'dunnett'
+]
+
+
+@dataclass
+class DunnettResult:
+    statistic: np.ndarray
+    pvalue: np.ndarray
+
+
+def dunnett(
+    *samples: npt.ArrayLike,
+    control: npt.ArrayLike,
+    alternative: Literal['two-sided', 'less', 'greater'] = "two-sided",
+    random_state: SeedType = None
+) -> DunnettResult:
+    """Dunnett's test: multiple comparisons of means against a control group.
+
+    Parameters
```

Comment:
I think this belongs somewhere to avoid confusion with the more recent tests.

```suggestion
    This is an implementation of Dunnett's original, single-step test as
    described in [1]_.

    Parameters
```